protocol/smart-batching.md

```diff
@@ -0,0 +1,84 @@
+# Smart Batching: Design Doc
+
+|                    |                                                    |
+| ------------------ | -------------------------------------------------- |
+| Author             | @jbearer (Espresso Systems)                        |
+| Created at         | 2025-03-25                                         |
+| Initial Reviewers  | Eli Haims, Sam McIngvale                           |
+| Need Approval From |                                                    |
+| Status             | In Review                                          |
+
+## Purpose
+
+Enabling plugin logic for application-specific sequencing or batching behavior without the need to fork the derivation pipeline.
+
+## Summary
+
+Enables chains to deploy a contract at the batch inbox address with custom logic for filtering
+batches. The derivation pipeline is modified to ignore batches where the contract reverts. This
+expands the design space for OP stack rollups running the canonical derivation pipeline.
+
+Example uses:
+
+- TEE-sequencer: L1 enforces that the rollup sequencer runs in a TEE, which enforces additional sequencing policies:
+    - Unichain wants to use a TEE for Flashblocks and verifiable priority ordering. Currently the derivation pipeline cannot check the TEE attestation, so the current implementation is no better than a trusted sequencer. This proposed change would provide an avenue to fix that. It will also be needed for the UVN.
+    - The Celo sequencer will run a TEE that first posts its transaction batches to a fast BFT consensus (run by Espresso Network) and only posts batches to the L1 that were already confirmed by this BFT consensus. This allows users and protocols (exchanges, bridges, etc) to obtain rollup soft confirmations (pre-L1 finality) from the fast BFT consensus that are more secure than confirmations coming from a regular centralized sequencer. Other OP stack chains are interested in adding this protective layer as well.
+- L1 directly verifies that the sequenced batches have been confirmed by an external layer (e.g. Espresso) instead of relying on a TEE.
```

Comment:
What does "confirmed" mean here? Are you talking about EspressoDA here or something else? Because one thing we are currently looking at for EigenDA is to verify a certificate of availability in the batcher inbox.